### PR TITLE
Implemented redux in app and into Auth page

### LINKED ISSUE #7: Use Redux on Auth
Improve Authentication pages / logic to work through Redux / Redux-Saga

### PR DESCRIPTION
**Pull request by dmitry-honcharuk**
Resolves #7

### CODE REVIEW

File: app/components/multi-column/MultiColumnLayout.jsx [hunk 1]
```diff
@@ -63,48 +64,49 @@ const Wrapper = styled.div`
...
+    onRightButtonClick: PropTypes.func.isRequired
+  };
+
+  static defaultProps = {
     currentState: STATES.LEFT
   };
 
-  changeState = newState => () => {
-    this.setState({ currentState: newState });
+  _handleButtonClick = state => () => {
```
Comment (Danteisabck) [inline app/components/multi-column/MultiColumnLayout.jsx]: Need comment

File: app/components/multi-column/MultiColumnLayout.jsx [hunk 2]
```diff
@@ -63,48 +64,49 @@ const Wrapper = styled.div`
...
+    if (state === STATES.LEFT) {
+      this.props.onLeftButtonClick();
+    } else if (state === STATES.RIGHT) {
+      this.props.onRightButtonClick();
+    }
   };
 
   _renderStateButton = (state, content) => (
```
Comment (Danteisabck) [inline app/components/multi-column/MultiColumnLayout.jsx]: Need comment , also describe params

File: app/components/pages/auth/Auth.jsx [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
+import { connect } from "react-redux";
+import AuthView from "./AuthView";
+import { changeAuthState } from "../../../actions/action-creators";
+import { AUTH_STATES } from "../../../reducers/auth-page";
+import { STATES as PAGE_STATES } from "../../multi-column/MultiColumnLayout";
 
-import MultiColumnLayout, {
-  STATES
-} from "../../multi-column/MultiColumnLayout";
+const getCurrentState = authState => {
```
Comment (Danteisabck) [inline app/components/pages/auth/Auth.jsx]: Need comment , also describe params

File: app/components/pages/auth/AuthView.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from "react";
+import LoginForm from "../../LoginForm";
+import RegistrationForm from "../../RegistrationForm";
+import ResetPasswordForm from "../../ResetPasswordForm";
+import MultiColumnLayout from "../../multi-column/MultiColumnLayout";
+
+const Auth = ({
```
Comment (Danteisabck) [inline app/components/pages/auth/AuthView.jsx]: Need comment , also describe params

**Comment by evgeniyannenkov** [comment]
Cool!

**Pull request merged**